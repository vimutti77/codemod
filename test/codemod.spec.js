const fs = require('fs')
const path = require('path')
const testUtils = require('jscodeshift/src/testUtils')

const testTransform = (name, { filePath } = {}) => {
  describe(name, () => {
    const root = path.join(__dirname, '..')
    const transformPath = `${root}/transforms/${name}.js`
    const fixturesPath = `${root}/test/fixtures/${name}`

    if (!fs.existsSync(fixturesPath)) {
      it.skip('create tests in ./test/fixtures/<transform-name>/<scenario-name>.input.js', () => {})
      return
    }

    fs.readdirSync(fixturesPath)
      .filter((filename) => filename.endsWith('.input.js'))
      .map((filename) => filename.replace('.input.js', ''))
      .forEach((id) => {
        const inputPath = `${fixturesPath}/${id}.input.js`
        const optionsPath = `${fixturesPath}/${id}.options.js`
        const displayPath = path.relative(root, inputPath)

        it(displayPath, () => {
          const transform = require(transformPath)
          const options = require(optionsPath)
          const input = fs.readFileSync(inputPath, 'utf8')
          testUtils.runSnapshotTest(transform, options, { path: filePath, source: input })
        })
      })
  })
}

// To write tests, create fixture files in the locations below and they
// will be discovered and run:
//
// ./test/fixtures/<transform-name>/<scenario-name>.input.js
// ./test/fixtures/<transform-name>/<scenario-name>.options.js

fs.readdirSync(path.resolve(__dirname, '../transforms'))
  .filter((filename) => filename.endsWith('.js'))
  .map((filename) => filename.replace('.js', ''))
  .forEach((transformName) => {
    testTransform(transformName)
  })
