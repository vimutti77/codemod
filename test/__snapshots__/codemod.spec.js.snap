// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`direct-import test\\fixtures\\direct-import\\default.input.js 1`] = `
"import Abc from \\"@mui/icons-material/Abc\\";
import PinDropIcon from \\"@mui/icons-material/PinDrop\\";
import Analytics from \\"@mui/icons-material/Analytics\\";
import AbcOutlinedIcon from \\"@mui/icons-material/AbcOutlined\\";"
`;
